build_docs.sh

```diff
@@ -0,0 +1,173 @@
+#!/bin/bash
+
+# Build the docs with docker!
+#
+# Step 1 is to build a docker image based on the asciidoctor image.
+# Step 2 is to translate the arguments that build_docs.pl supports into
+# a list of arguments to be passed to start the docker container and a
+# list of arguments to be passed to the build_docs.pl process that is
+# started in the docker container.
+# Step 3 is to start the docker container. We start it in such a way
+# that is *should* remove itself when it is done.
+
+set -e
+
+function desymlink() {
+  FILE="$1"
+  while [ -h "$FILE" ] ; do
+    ls=$(ls -ld "$FILE")
+    # Drop everything prior to ->
+    link=$(expr "$ls" : '.*-> \(.*\)$')
+    if expr "$link" : '/.*' > /dev/null; then
+      FILE="$link"
+    else
+      FILE=$(dirname "$FILE")/"$link"
+    fi
+  done
+  echo $FILE
+}
+
+function to_absolute_path() {
+  FILE=$(desymlink "$1")
+  cd "$(dirname "$FILE")"
+  echo "$(pwd -P)/$(basename "$FILE")"
+}
+
+# Find the root of a git repo for a given directory if
+# possible, otherwise returning the directory itself.
+function find_git_repo_root() {
+  cd $1
+  while [ ! -d ".git" ]; do
+    if [ "$PWD" = "/" ]; then
+      # Not in a git repo
+      echo "$1"
+      return
+    fi
+    cd ..
+  done
+  echo "$PWD"
+}
+
+DIR="$(dirname "$(to_absolute_path "$0")")"
+
+DOCKER_RUN_ARGS=()
+
+# Remove the container when we are done with it.
+DOCKER_RUN_ARGS+=('--rm')
+
+# Make sure we create files as the current user because that is what
+# folks that use build_docs.pl expect.
+DOCKER_RUN_ARGS+=('--user' "$(id -u):$(id -g)")
+
+# Running read-only with a proper tmp directory gives us a little
+# performance boost and it is simple enough to do.
+DOCKER_RUN_ARGS+=('--read-only')
+DOCKER_RUN_ARGS+=('--tmpfs' '/tmp')
+
+# Mount the docs build code so we can run it!
+DOCKER_RUN_ARGS+=('-v' "$DIR:/docs_build:cached")
+
+# Seccomp adds a *devestating* performance overhead if you happen
+# to have it installed.
+DOCKER_RUN_ARGS+=('--security-opt' 'seccomp=unconfined')
```

Comment:
See the commit message for this one. I don't like it, but without it my laptop *hates* this change. I've spent a few days looking into it and I'm cutting my losses for now.